### PR TITLE
Dataset imbalance ,better data augmentation for Audio to image, inference

### PR DESCRIPTION
**Pull request by Flippchen**
- Implementing data augmentation with AugmentMelSpectrogram in data_augmentation.py.
- Integrating augmentation into the main script via the BirdDataset class.
- Balancing the dataset using oversampling or undersampling in balance.py.
-  Added CSV for balanced dataset 
- Computing class weights for the loss function with calculate_class_weights in class_weights.py.
- Adding visualization functions in plot_class_occurences.py and plot_waveform.py
- Added standalone inference

### CODE REVIEW

**Comment by Flippchen** [comment]
As I introduced the "balanced" Dataset. Should we recalculate the train/val/test and local split? I set the target samples for each class to 100.

File: src/audio_to_image_model/audio_to_iamge_standalone_predict.py [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+    return result_df
+
+
+if __name__ == '__main__':
+    # Set parameters here
+    TEST_FILE_PATH = r"D:\kaggle_competition\birdclef-2023\test_soundscapes"
+    device = "cuda" if torch.cuda.is_available() else "cpu"
+
+    # Load your model and preprocessing function here
+    preprocess_fn = preprocess_audio
```
Comment (anwielts) [inline src/audio_to_image_model/audio_to_iamge_standalone_predict.py:76]: I do not get what this is doing. As far as I can see the function is not used?

**Review by anwielts — APPROVED** [review]
Only one comment, can be merged nevertheless

File: src/audio_to_image_model/audio_to_iamge_standalone_predict.py (hunk 1 above)
Comment (Flippchen) [reply src/audio_to_image_model/audio_to_iamge_standalone_predict.py:76]: thank you, I have overlooked that

**Pull request merged**